matter_server/server/device_controller.py

```diff
@@ -403,27 +468,71 @@ async def subscribe_node(self, node_id: int) -> None:
             raise NodeNotExists(
                 f"Node {node_id} does not exist or has not been interviewed."
             )
-        assert node_id not in self._subscriptions, "Already subscribed to node"
-        node_logger = LOGGER.getChild(str(node_id))
+
+        node_logger = LOGGER.getChild(f"[node {node_id}]")
         node_logger.debug("Setting up subscriptions...")
+        node_lock = self._get_node_lock(node_id)
+        # check if we already have an subscription for this node,
+        # if so, we need to unsubscribe first because a device can only maintain
+        # a very limited amount of concurrent subscriptions.
+        if sub := self._subscriptions.pop(node_id, None):
+            async with node_lock:
+                node_logger.info("Unsubscribing from existing subscription.")
```

Comment:
```suggestion
                node_logger.debug("Unsubscribing from existing subscription.")
```

Reply:
I changed this on purpose so I can track it for a while in user logs. I'll change it back to debug in a while